### PR TITLE
Vessel events layer part 2

### PR DESCRIPTION
**Pull request by nerik**
Full parse data from API + time filtering

### CODE REVIEW

File: src/layer-composer/layer-composer.ts [hunk 1]
```diff
@@ -21,7 +21,6 @@ class LayerComposer {
   sprite: string
   generators: { [key: string]: any }
   latestGenerated: any
-  globalGeneratorConfig: GlobalGeneratorConfig
```
Comment (j8seangel) [inline src/layer-composer/layer-composer.ts:24]: isn't this used anymore?

File: src/layer-composer/generators/vessel-events/vessel-events.ts [hunk 2]
```diff
@@ -106,27 +158,40 @@ const getEncounterAuthColor = (authorizationStatus: AuthorizationOptions) => {
...
+
+  if (!trackEvents) return featureCollection
+
+  featureCollection.features = trackEvents.map((event: RawEvent) => {
+    const authorized = event.encounter && event.encounter.authorized === true
+    const authorizationStatus = event.encounter
+      ? event.encounter.authorizationStatus
+      : ('unmatched' as AuthorizationOptions)
+
+    const lng = event.position.lng || event.position.lon || 0
```
Comment (j8seangel) [inline src/layer-composer/generators/vessel-events/vessel-events.ts:175]: do you know where the difference nomenclature is coming from? I'd like to normalize and use always the same literal

File: src/layer-composer/generators/vessel-events/vessel-events.ts [hunk 3]
```diff
@@ -62,27 +109,32 @@ class VesselsEventsGenerator {
...
-    lng: number
+  position: {
+    lng?: number
+    lon?: number
     lat: number
   }
+  start: number
+  encounter?: {
+    authorized: boolean
+    authorizationStatus: AuthorizationOptions
```
Comment (j8seangel) [inline src/layer-composer/generators/vessel-events/vessel-events.ts:136]: this is directly related with the API response, do you think worth it to keep it synced using the same types ? however how to reuse types across projects is something we have to discuss I'd like to open the conversation.

File: src/layer-composer/generators/vessel-events/vessel-events.ts [hunk 4]
```diff
@@ -62,27 +109,32 @@ class VesselsEventsGenerator {
...
+      id: config.id,
+      sources: this._getStyleSources(config),
+      layers: this._getStyleLayers(config),
     }
   }
 }
 
 export default VesselsEventsGenerator
 
 type AuthorizationOptions = 'authorized' | 'partially' | 'unmatched'
```
Comment (j8seangel) [inline src/layer-composer/generators/vessel-events/vessel-events.ts:123]: same here related on how to reuse types

**Review by j8seangel — APPROVED** [review]
Looks great! I love how easy is to render the layer across projects just a few comments about how to scale this but ready to merge for now!

File: src/layer-composer/layer-composer.ts (hunk 1 above)
Comment (nerik) [reply src/layer-composer/layer-composer.ts:24]: it's not

File: src/layer-composer/generators/vessel-events/vessel-events.ts (hunk 2 above)
Comment (nerik) [reply src/layer-composer/generators/vessel-events/vessel-events.ts:175]: The API uses "lon", while the original carrier portal uses "lng". 
I guess I will use "lon" ang be done with it.

**Pull request merged**